README.md

```diff
@@ -11,51 +11,61 @@ related-issues: (will contain links to implementation PRs)
+This makes it much easier to contribute new packages packages, since there's no more guessing needed as to where the package should go, both in the ad-hoc directory categories and in `pkgs/top-level/all-packages.nix`.
+
 
 # Motivation
 [motivation]: #motivation
 
-- (Especially new) package contributors are having a hard time figuring out which files to add and edit. These are very common questions:
-  - Which directory should my package definition go in? What are all the categories and do they matter? What if the package has multiple matching categories?
-  - Why can't I build my package after adding the package file? [introduced to all-packages.nix] Where in all-packages.nix should my package go?
-- Figuring out where an attribute is defined is very tricky:
+- (Especially new) package contributors are having a hard time figuring out which files to add or edit. These are very common questions:
+  - Which directory should my package definition go in?
+  - What are all the categories and do they matter?
+  - What if the package has multiple matching categories?
+  - Why can't I build my package after adding the package file?
+  - Where in all-packages.nix should my package go?
+- Figuring out where an attribute is defined is a bit tricky:
   - First one has to find the definition of it in all-packages.nix to see what file it refers to
-    - Especially on GitHub this is even more problematic, as the `all-packages.nix` file is [too big to be displayed by GitHub](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/top-level/all-packages.nix)
+    - On GitHub this is even more problematic, as the `all-packages.nix` file is [too big to be displayed by GitHub](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/top-level/all-packages.nix)
   - Then go to that file's definition, which takes quite some time for navigation (unless you have a plugin that can jump to it directly)
 - `all-packages.nix` frequently causes merge conflicts. It's a point of contention for all new packages
 
 # Detailed design
 [design]: #detailed-design
 
-All attributes at the root of nixpkgs (`pkgs.<name>`) whose definition is of the form
+For all attributes at the root of nixpkgs `pkgs.<name>` which:
+1. Are defined in `pkgs/top-level/all-packages.nix`
+  (necessary so that the overlay containing the automatically discovered packages can be ordered directly before the `all-packages.nix` overlay without changing any behavior)
+2. Are defined to be equal to `pkgs.callPackage <path> { }` such that all transitively referenced paths from the default Nix file of `<path>`:
+  1. Are under the directory of `<path>`
+    (necessary so that moving these files to a new directory doesn't break references in this package)
+  2. Are not referenced from any paths outside of these transitive references
+    (necessary so that moving these files to a new directory doesn't break references in other packages)
```

Comment:
We discussed this [in yesterdays NAT meeting](https://discourse.nixos.org/t/2022-11-28-nixpkgs-architecture-team-meeting-19/23572#relative-file-references-8). We concluded that this is a good idea, but it's hard to understand what's written here, it should be reworded to something like "Limited to packages that can be moved without updating any file references"

Reply:
The tooling could also be regularly ran in CI, giving warnings for all packages that can't be moved automatically, along with an indication for why it couldn't be moved automatically. This is especially a list of special cases we need to take a look at then

Reply:
Updated the RFC to simplify the wording of this part